server/src/main/java/org/elasticsearch/common/util/CuckooFilter.java

```diff
@@ -95,6 +97,27 @@
         this.fingerprintMask = (0x80000000 >> (bitsPerEntry - 1)) >>> (Integer.SIZE - bitsPerEntry);
     }
 
+    CuckooFilter(CuckooFilter other) {
+        this.numBuckets = other.numBuckets;
+        this.bitsPerEntry = other.bitsPerEntry;
+        this.entriesPerBucket = other.entriesPerBucket;
+        this.count = other.count;
+        this.evictedFingerprint = other.evictedFingerprint;
+        this.rng = other.rng;
+        this.fingerprintMask = other.fingerprintMask;
+
+        // This shouldn't happen, but as a sanity check
+        if (numBuckets * entriesPerBucket > Integer.MAX_VALUE) {
+            throw new IllegalArgumentException("Attempted to create [" + numBuckets * entriesPerBucket
+                + "] entries which is > Integer.MAX_VALUE");
+        }
+        // TODO this is probably super slow, but just used for testing atm
```

Comment:
Maybe this warning should be in aJavaDoc on this constructor since it indicates this constructor should only be used in tests?

Reply:
:+1: